zebra-network/src/peer_set/set/tests/prop.rs

```diff
@@ -90,17 +94,85 @@ proptest! {
             Ok::<_, TestCaseError>(())
         })?;
     }
+
+    /// Test if requests are broadcasted to the right number of peers.
+    #[test]
+    fn broadcast_to_peers(
+        total_number_of_peers in (1..100usize)
+    ) {
+        // Get a dummy block hash to help us construct a valid request to be broadcasted
+        let block: block::Block = zebra_test::vectors::BLOCK_MAINNET_10_BYTES
+            .zcash_deserialize_into()
+            .unwrap();
+        let block_hash = block::Hash::from(&block);
+
+        // Start the runtime
+        let runtime = zebra_test::init_async();
+        let _guard = runtime.enter();
+
+        let peer_versions = vec![CURRENT_NETWORK_PROTOCOL_VERSION; total_number_of_peers];
+        let peer_versions = PeerVersions {
+            peer_versions,
+        };
+
+        // Get peers and handles
+        let (discovered_peers, mut handles) = peer_versions.mock_peer_discovery();
+        let (minimum_peer_version, _best_tip_height) =
+            MinimumPeerVersion::with_mock_chain_tip(Network::Mainnet);
+
+        // Build a peerset
+        runtime.block_on(async move {
+            let (mut peer_set, _peer_set_guard) = PeerSetBuilder::new()
+                .with_discover(discovered_peers)
+                .with_minimum_peer_version(minimum_peer_version.clone())
+                .build();
+
+            // Get the total number of active peers
+            let total_number_of_active_peers = check_if_only_up_to_date_peers_are_live(
+                &mut peer_set,
+                &mut handles,
+                CURRENT_NETWORK_PROTOCOL_VERSION,
+            )?;
+
+            // The number of active peers is never greater than the total
+            prop_assert!(total_number_of_peers >= total_number_of_active_peers);
+
+            // Get the number of peers to broadcast
+            let number_of_peers_to_broadcast = peer_set.number_of_peers_to_broadcast();
+
+            // The number of peers to broadcast is never greater than the actives
+            prop_assert!(total_number_of_active_peers >= number_of_peers_to_broadcast);
```

Comment:
This helps make sure that:
- we never get zero peers for a broadcast, because that's wrong
- we always leave some peers for other requests, because it's a performance issue if we don't

```suggestion
            // The number of peers to broadcast should be at least 1,
            // and if possible, it should be less than the number of ready peers.
            // (Since there are no requests, all active peers should be ready.)
            prop_assert!(number_of_peers_to_broadcast >= 1);
            if total_number_of_active_peers > 1 {
                prop_assert!(number_of_peers_to_broadcast < total_number_of_active_peers);
            }
```

Reply:
Added this one here https://github.com/ZcashFoundation/zebra/pull/3284/commits/9700efdc725960909bc34475ae3aa42d806c6f1a, thanks.